documentation/design/custom-schema.adoc

```diff
@@ -0,0 +1,25 @@
+= Support for Custom Schema
+
+== The problem
+Currently, Timbuctoo only supports auto-generated Schema.
+Users do not have the ability to extend on the auto-generated schema for their datasets.
+
+== The result
+Users are able to add custom schema to the dataset's auto-generated schema
```

Comment:
Another important characteristic is that users should only be able to add to the schema. So the merge should only expand the schema:

 - A property may be added, but not removed
 - A property may receive a new union type, but not have one removed
 - A singular property may be turned into a list, but not the other way around
 - We do not deal with nullability since we're not asserting that anyway. (it should be removed from the graphql api since it cannot be migrated away from in a backwards compatible manner)

Reply:
deprecation should be allowed though.